### PR TITLE
fixes #1723: Github Actions Supported

### LINKED ISSUE #1723: Migrate from travis-ci.org before the end of May 2021
From travis-ci.org:

> Please be aware travis-ci.org will be shutting down by end of May 2021. Please consider migrating to travis-ci.com.

It would probably be little-to-no effort to migrate to travis-ci.com. Open source projects still run free on travis-ci.com. I haven't looked at how much configuration changes might be needed, or how much lede time might be required to make this change.

If we are unhappy with Travis CI generally, I would be happy to create configuration for GitHub Actions if we have interest in trying that. I would not be able to do that before June 2021, however.

### PR DESCRIPTION
**Pull request by darless1**
- This is to switch from Travis CI which is shutting down the org version,
  and the .com version, a free account is limited.
- Switch test environment variable from TRAVIS to CI to be more generic.

### CODE REVIEW

**Comment by darless1** [comment]
Associated to #1723

**Comment by darless1** [comment]
Please notes - that I removed checking of jruby-9.2.9.0 since it's been failing for some time for other PRs as well without changes.

https://travis-ci.org/github/gollum/gollum/jobs/774413722
```
Retrying fetcher due to error (2/4): Bundler::Fetcher::CertificateFailureError Could not verify the SSL certificate for https://rubygems.org/.
There is a chance you are experiencing a man-in-the-middle attack, but most likely your system doesn't have the CA certificates needed for verification. For information about OpenSSL certificates, see http://bit.ly/ruby-ssl. To connect without using SSL, edit your Gemfile sources and change 'https' to 'http'.
Retrying fetcher due to error (3/4): Bundler::Fetcher::CertificateFailureError Could not verify the SSL certificate for https://rubygems.org/.
There is a chance you are experiencing a man-in-the-middle attack, but most likely your system doesn't have the CA certificates needed for verification. For information about OpenSSL certificates, see http://bit.ly/ruby-ssl. To connect without using SSL, edit your Gemfile sources and change 'https' to 'http'.
Retrying fetcher due to error (4/4): Bundler::Fetcher::CertificateFailureError Could not verify the SSL certificate for https://rubygems.org/.
There is a chance you are experiencing a man-in-the-middle attack, but most likely your system doesn't have the CA certificates needed for verification. For information about OpenSSL certificates, see http://bit.ly/ruby-ssl. To connect without using SSL, edit your Gemfile sources and change 'https' to 'http'.
Could not verify the SSL certificate for https://rubygems.org/.
There is a chance you are experiencing a man-in-the-middle attack, but most
likely your system doesn't have the CA certificates needed for verification. For
information about OpenSSL certificates, see http://bit.ly/ruby-ssl. To connect
without using SSL, edit your Gemfile sources and change 'https' to 'http'.
The command "eval bundle install --jobs=3 --retry=3 " failed. Retrying, 2 of 3.
```

If this is truly necessary to test with, then I can look into how to get it to work. The failures is seen both with Travis and Github Actions.

File: test/test_migrate.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def load_script(**args)
   end  
 end
 
-unless ENV['TRAVIS']
+unless ENV['CI']
```
Comment (darless1) [inline test/test_migrate.rb:47]: This environment is in the default environment variables for the travis CI and github actions.

**Comment by benjaminwil** [comment]
@darless1 I think that the JRuby failures are due to an issue with an older version of `jruby-openssl`.

Could you update the JRuby version to `9.2.18.0` and see if it now passes? I did this for travis-ci.org and I was able to get a pass. (See #1730.)

**Comment by darless1** [comment]
https://github.com/darless1/gollum/actions/runs/933997416

The latest run with jruby-9.2.18.0 worked without issues.

**Comment by benjaminwil** [comment]
Thank you!

@dometto @bartkamphorst How do you feel about migrating from Travis to GitHub Actions? You can see @darless1's successful run [here](https://github.com/darless1/gollum/runs/2815385151) as an example.

After merging, I believe all that would be necessary is configuring GitHub Actions as desired for the `gollum` org.

**Comment by bartkamphorst** [comment]
Thanks for your work on this @darless1 ! I'm in favor of making the switch. @dometto ?

**Comment by dometto** [comment]
Sorry for the extremely slow reply, and thanks to everyone for helping out with this! I'm in favour of migrating to Github Actions, too.

**Comment by bartkamphorst** [comment]
>  I believe all that would be necessary is configuring GitHub Actions as desired for the gollum org.

What does this entail @benjaminwil ?

**Comment by benjaminwil** [comment]
@bartkamphorst Looks like I was wrong! Merging this code "just worked" and GitHub actions is enabled. 🎊

**Comment by bartkamphorst** [comment]
Very cool. Happy to have made the switch. Thanks @benjaminwil for flagging the issue and @darless1 for the PR!

**Pull request merged**